crates/uv/src/requirements.rs

```diff
@@ -583,118 +585,88 @@ impl NamedRequirements {
             });
         }
 
-        // Otherwise, download and/or extract the source archive.
-        if Scheme::parse(requirement.url.scheme()) == Some(Scheme::File) {
-            let path = requirement.url.to_file_path().map_err(|()| {
-                anyhow::anyhow!("Unable to convert file URL to path: {requirement}")
+        // Download the archive and attempt to infer the package name from the archive contents.
+        let source = download_archive(&requirement.url, cache, client)
```

Comment:
Should we just be extracting the files of interest here? Or is that not worthwhile because we extract the whole thing later for installation and we're caching extracting for the duration of this process?

Reply:
Yeah, this is very wasteful (we don't cache this, we have to re-extract later, but this is only for HTTPS requirements that don't reference a specific built file, I don't know that this really even happens in practice). I could change it to only extract the files of interest. The downside is that if we _do_ want to use PEP 517 hooks (which we're "supposed to"), then we need to do the full extraction anyway.

Reply:
(Aside) It's just really hard to cache here in a way that can be reused later because we don't have a `Dist`, and all the caching layers require `Dist`. Like, ideally, we could just use the `DistributionDatabase` for this, but that requires `Dist` everywhere.

Reply:
I'd be okay with only extracting a subset of files in a future optimization. I think regardless we'd only want to use a PEP 517 hook _after_ our fast heuristics are exhausted, right?

What are we missing from a `Dist`? Just the name? Can we put it in the cache after we've acquired the name?

Reply:
In theory yes but executing on that is pretty tricky given the way the code is structured. I plan to try it...

The other option is that we could _remove_ the package name from the cache, and just use the URL as a key. This would make things much easier. (And it would mean we could _read_ from the cache here, not just write.) The main downside (AFAICT) is that it would (1) break the cache so we'd need to bump version, and (2) break `uv cache clean flask` so we'd need to figure out how to support that (it's possible).

Reply:
I think it's doable but it's _much_ easier if we remove the package name from the cache entries.

Reply:
Ok, first I'm gonna make it so we don't unnecessarily extract... Then I will look into caching these...